### ISSUE TITLE
No history for pure tofi?

### ISSUE THREAD

**Issue by b1rger** [philj56/tofi#57]
I use tofi together with `pass` for selecting and copying passwords. It would be great if tofi would remember the passwords I used in the past and puts them at the beginning of the selection. This seems to be implemented for `tofi-run` and `tofi-drun` with the history files in `$XDG_STATE_HOME/tofi-history` and `$XDG_STATE_HOME/tofi-drun-history` - would it be possible to implement that feature for `tofi` (i.e. by setting a path in the configuration file)?

**Comment by philj56** [comment]
I did think about that when I did the history stuff, and I still think it shouldn't be on by default for a reason similar to that - people probably won't want plaintext passwords or other sensitive info being saved to the disk without them knowing, if they were to actually put that text in tofi.

I don't think there's any reason not to make it an option though. My only thought is whether to make it a different option, like `histfile-path=<path>`, or overload the existing `history` option behaviour, to get `history=<true|false|path>`, or something else. Needs a little bit of thought.

**Comment by philj56** [comment]
Alright, I've added a `--history-file=<path>` option in d04ca544d6845fc655cd4ba55e134de6bead6a2d & 9ceceeb59a95919e5fe09338dc27e4fac0b5aa24. This is effective in all modes, not just in pure tofi. Effectively then, `tofi-run` is now identical to
```sh
tofi-compgen | tofi --history-file=$XDG_STATE_HOME/tofi-history
```
which is nice, as building your own modes (such as your `pass` mode) is just as simple.

One caveat currently is that the items passed to tofi are sorted alphabetically, prior to history sorting. I'm not sure whether this is really a problem though - I can't think of any cases where you'd both want history sorting and care about the order of non-history-sorted items. Still, it violates the principle of least surprise, so I'll look at fixing that.

**Comment by philj56** [comment]
I've fixed the extra sorting in 5da0579864a13b11bf0808f7f69f7b22f18054b1, so now everything should be working as expected.
